n3fit/src/n3fit/vpinterface.py

```diff
@@ -46,94 +47,34 @@
+    however, since there is no grid, the CV has to be computed manually"""
 
-    Parameters
-    ----------
-        pdf_models: :py:class:`n3fit.backends.MetaModel` (or list thereof)
-            PDF trained with n3fit, x -> f(x)_{i} where i are the flavours in the evol basis
-        fit_basis: list(dict)
-            basis of the training, used for reporting
-        name: str
-            name of the N3PDF object
-    """
+    def error_members(self):
+        return self.data
 
-    def __init__(self, pdf_models, fit_basis=None, name="n3fit"):
-        self.fit_basis = fit_basis
-        self.basis = check_basis("evolution", EVOL_LIST)["basis"]
+    def central_value(self):
+        return np.mean(self.data, axis=0)
 
-        if isinstance(pdf_models, Iterable):
-            self._models = pdf_models
-        else:
-            self._models = [pdf_models]
 
-        super().__init__(name)
-        # The number of members will be minimum 2, needed for legacy compatibility
-        # Note that the member 0 does not exist as an actual model as it corresponds
-        # to the average of all others
-        self._info = {"ErrorType": "replicas", "NumMembers": len(self._models) + 1}
+class N3LHAPDFSet(LHAPDFSet):
+    """Extension of LHAPDFSet using n3fit models"""
 
-    @property
-    def stats_class(self):
-        """The stats class of N3PDF is always MCStats"""
-        return MCStats
-
-    def load(self):
-        """Many vp functions ask for a LHAPDF pdf
-        from nnpdflib, this class fakes it until a time in which vp is free from C++
-        """
-        return self
-
-    def get_member_model(self, replica):
-        """Return the member corresponding to the given replica"""
-        return self._models[replica - 1]
-
-    def get_nn_weights(self):
-        """Outputs all weights of the NN as numpy.ndarrays"""
-        return [model.get_weights() for model in self._models]
-
-    def get_preprocessing_factors(self, replica=None):
-        """Loads the preprocessing alpha and beta arrays from the PDF trained model.
-        If a ``fit_basis`` given in the format of ``n3fit`` runcards is given it will be used
-        to generate a new dictionary with the names, the exponent and whether they are trainable
-        otherwise outputs a Nx2 array where [:,0] are alphas and [:,1] betas
-        """
-        # If no replica is explicitly requested, get the preprocessing layer for the first model
-        if replica is None:
-            replica = 1
-        preprocessing_layers = self.get_member_model(replica).get_layer_re(r"pdf_prepro_\d")
-        if len(preprocessing_layers) != 1:
-            # We really don't want to fail at this point, but print a warning at least...
-            log.warning("More than one preprocessing layer found within the model!")
-        preprocessing_layer = preprocessing_layers[0]
+    def __init__(self, name, pdf_models):
+        log.info("Creating LHAPDF-like n3fit PDF")
```

Comment:
Not a huge fan of printing to screen when objects are created...

But at least:

```suggestion
        log.debug("Creating LHAPDF-like n3fit PDF")
```